In [1]:
import pandas as pd

from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

from joblib import dump

In [2]:
df = pd.read_csv('train_data.csv')

In [3]:
df.loc[df['last_price'] > 5650000, 'price_class'] = 1
df.loc[df['last_price'] <= 5650000, 'price_class'] = 0

In [4]:
features = df.drop(['last_price', 'price_class'], axis=1)
target = df['price_class']

In [5]:
model = LogisticRegression(random_state=12345)

In [6]:
model.fit(features, target)

/Users/akrisanov/opt/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)


LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='warn', n_jobs=None, penalty='l2',
                   random_state=12345, solver='warn', tol=0.0001, verbose=0,
                   warm_start=False)

In [7]:
predictions = model.predict(features)

In [8]:
accuracy = accuracy_score(target, predictions)

In [9]:
accuracy

0.8808314087759815

Accuracy:

* Обучающая выборка: 0.8808314087759815
* Тестовая выборка: 0.8890086206896551 – Accuracy ниже. Зато модель просто летает. 

In [10]:
dump(model, 'logistic_regression.joblib')

['logistic_regression.joblib']